purge-cluster.yml

```diff
@@ -2,42 +2,193 @@
-    failed_when: false
+  - name: stop ceph-osd with systemd
+    service:
+      name: ceph-osd@{{item}}
+      state: stopped
+      enabled: no
+    with_items: "{{ osd_ids.stdout_lines }}"
+    when:
+      ansible_distribution != 'Ubuntu' and
+      ceph_stable_release == 'infernalis' and
+      osd_group_name in group_names
 
-  - name: purge remaining data
-    command: ceph-deploy purgedata {{ ansible_fqdn }}
-    delegate_to: 127.0.0.1
+  - name: stop ceph mons with systemd
+    service: 
+      name: ceph-mon@{{ ansible_hostname }} 
+      state: stopped
+      enabled: no
+    when:
+      ansible_distribution != 'Ubuntu' and
+      ceph_stable_release == 'infernalis' and
+      mon_group_name in group_names 
 
-- hosts:
-  - osds
+  - name: stop ceph mdss with systemd
+    service: 
+      name: ceph-mds@{{ ansible_hostname }} 
+      state: stopped
+    when:
+      ansible_distribution != 'Ubuntu' and
+      ceph_stable_release == 'infernalis' and
+      mds_group_name in group_names 
 
-  vars:
-    devices: [ '/dev/sdb', '/dev/sdc', '/dev/sdd', '/dev/sde', '/dev/sdf' ]
-    partitions: [ '1', '2', '3' ]
+# before infernalis
+  - name: stop ceph osds
+    command: service ceph stop osd
+    when:
+      ansible_distribution != 'Ubuntu' and
+      osd_group_name in group_names and
+      ceph_stable_release != 'infernalis'
 
-  tasks:
+  - name: stop ceph mons
+    command: service ceph stop mon
+    when:
+      ansible_distribution != 'Ubuntu' and
+      mon_group_name in group_names and
+      ceph_stable_release != 'infernalis'
 
-  - name: disk zap
-    command: /usr/sbin/sgdisk --zap-all --clear --mbrtogpt -g -- {{ item }}
-    with_items: devices
+  - name: stop ceph mdss
+    command: service ceph stop mds
+    when:
+      ansible_distribution != 'Ubuntu' and
+      mds_group_name in group_names and
+      ceph_stable_release != 'infernalis'
+
+# Ubuntu 14.04
+  - name: stop ceph osds on ubuntu
+    command: stop ceph-osd-all
     failed_when: false
+    when:
+      ansible_distribution == 'Ubuntu' and
+      osd_group_name in group_names
 
-  - name: disk zap again
-    command: /usr/sbin/sgdisk --zap-all --clear --mbrtogpt -g -- {{ item }}
-    with_items: devices
+  - name: stop ceph mons on ubuntu
+    command: stop ceph-mon-all
+    failed_when: false
+    when:
+      ansible_distribution == 'Ubuntu' and
+      mon_group_name in group_names
+  
+  - name: stop ceph mdss on ubuntu
+    command: stop ceph-mds-all
     failed_when: false
+    when:
+      ansible_distribution == 'Ubuntu' and
+      mds_group_name in group_names
+
+# rc is 2 if file not found, so no mount point, so no error
+  - name: get osd data mount points
+    shell: ls /var/lib/ceph/osd
+    register: mounted_osd
+    changed_when: false
+    failed_when: mounted_osd.rc != 0 and mounted_osd.rc != 2
+    when:
+     osd_group_name in group_names
+
+  - name: umount osd data partition
+    shell: umount /var/lib/ceph/osd/{{ item }}
```

Comment:
Problem with mount module is that it operates on /etc/fstab, not current system state. So for me it is not useful in this case as all osds are mounted dynamically, without any change is fstab.